playbooks/adhoc/s3_registry/s3_registry.yml

```diff
@@ -5,9 +5,20 @@
 #
 # The AWS access/secret keys should be the keys of a separate user (not your main user), containing only the necessary S3 access role.
 # The 'clusterid' is the short name of your cluster.
+#
+# This playbook should be limited to one openshift master
 
 - hosts: tag_clusterid_{{ clusterid }}:&tag_host-type_openshift-master
   remote_user: root
+  tasks:
+  - add_host:
+      name: hostvars[item].name
+      group: oo_first_master
+    with_items: "{{ play_hosts[0] }}"
```

Comment:
The way you are currently doing this, you are running the add_host task against all hosts specified. In order to limit this to a single execution you could do the following:
```
- hosts: localhost
  gather_facts: no
  become: no
  tasks:
  - add_host:
      name: "{{ (groups['tag_clusterid_' ~ clusterid] | union(groups['tag_host-type_openshift-master']))[0] }}"
      group: oo_first_master
```
or:
```
- hosts: tag_clusterid_{{ clusterid }}:&tag_host-type_openshift-master
  remote_user: root
  become: no
  tasks:
  - add_host:
      name: "{{ inventory_hostname }}"
      group: oo_first_master
    run_once: yes
```

As an aside, the way you are specifying the name attribute to add_host, it will actually set the name to 'hostvars[item].name', you will need to use an explicit jinja2 template there to evaluate the value.

The with_items on the other hand should use an implicit jinja2 template, but you need to pass it a list, so you could do this instead:
```
with_items: [play_hosts[0]] 
```